packages/react-components/react-tabster/src/hooks/useGetKeyboardNavigation.ts

```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import { useKeyborgRef } from './useKeyborgRef';
+
+/**
+ * Instantiates [keyborg](https://github.com/microsoft/keyborg) and checks if the user is navigating with the keyboard.
+ * @returns {() => boolean}
+ */
+export function useGetKeyboardNavigation() {
```

Comment:
I suggest we make a minor adjustment to the hook for enhanced clarity and adaptability in the future:



- Option 1, we could change the hook's name to `useIsNavigatingWithKeyboard` to more accurately reflect its specific function.



```ts
export function useIsNavigatingWithKeyboard() {
  const keyborgRef = useKeyborgRef();

	return React.useCallback(() => {
		return keyborgRef.current?.isNavigatingWithKeyboard() ?? false;
	}, [keyborgRef]);
}
```


- Option 2 involves renaming the hook to `useKeyboardNavigation`, which is a bit more generic. This approach would involve returning an object with named methods rather than just a single function. This change would facilitate the addition of other keyboard navigation features later on.



```ts
export function useKeyboardNavigation() {
  const keyborgRef = useKeyborgRef();

	return {
		isNavigatingWithKeyboard: React.useCallback(() => {
			return keyborgRef.current?.isNavigatingWithKeyboard() ?? false;
		}, [keyborgRef]);
	}
```


I lean towards the Option 2 as it offers greater flexibility.

Reply:
Thanks for suggestion. I will pick option 1, as other hooks in tabster are very granular, there are hooks like `useKeyboardNavAttribute`, `useOnKeyboadNavigationChange` having generic `useKeyboardNavigation` won't fit here very well 🙂